### PR TITLE
Add tests to sql-users custom resource

### PR DESCRIPTION
**Pull request by cat5inthecradle**
@Suresh this is merging into your branch.

Adding some tests, and appeasing the linter.

Run tests with `npm run test`

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: aws/cloudformation/lambdas/sql-user/cfn-response.js [hunk 1]
```diff
@@ -4,17 +4,24 @@
...
-exports.send = function(event, context, responseStatus, responseData, physicalResourceId, noEcho) {
-
+exports.send = function (
+  event,
+  context,
+  responseStatus,
+  responseData,
+  physicalResourceId,
+  noEcho
+) {
```
Comment (cat5inthecradle) [inline aws/cloudformation/lambdas/sql-user/cfn-response.js:14]: our pre-commit hook linter `run_eslint_apps` in "hooks.rb" is not happy with these files, so I ran it with the `--fix` option to auto-fix these files.

**Comment by cat5inthecradle** [comment]
I could see adding some unhappy path tests to cover the likely error scenarios, but ultimately any thrown exception should bounce back to Cloudformation.

File: aws/cloudformation/components/database.yml.erb [hunk 2]
```diff
@@ -185,6 +217,40 @@
...
+    DependsOn: [<%=DB_INSTANCE_RANGE.map {|i| "Aurora#{i}"}.join(',')%>]
+<%  end -%>
+    Properties:
+      ServiceToken: !GetAtt SQLUserLambda.Arn
+      DBServerHost: !GetAtt AuroraCluster.Endpoint.Address
+      # Credentials for SQL admin user that will create/update/delete this SQL user resource.
+      DBCredentialAdminSecret: !Ref DatabaseSecretAdmin
+      # Credentials (username and password) for the SQL User this Resource is provisioning.
+      DBCredentialSecret: !Ref DatabaseSecretWriter
+      Privileges:
```
Comment (pablo-code-org) [inline aws/cloudformation/components/database.yml.erb]: nit: can this be sorted or condensed like this?

```
    Privileges: [
      'SELECT', 'INSERT', 'UPDATE', 'DELETE', 'CREATE', 'DROP', 'REFERENCES', 'INDEX',
      'ALTER', 'CREATE TEMPORARY TABLES', 'LOCK TABLES', 'EXECUTE', 'CREATE VIEW',
      'SHOW VIEW', 'CREATE ROUTINE', 'ALTER ROUTINE', 'EVENT', 'TRIGGER'
    ]
```

File: aws/cloudformation/components/database.yml.erb [hunk 3]
```diff
@@ -110,6 +110,38 @@
...
+      Path: /
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'
+        - 'arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole'
+      Policies:
+        # TODO: Provision a Policy and use it in both the SQL User and DBProxy Roles.
+        - PolicyName: GetDBCredentials
+          PolicyDocument:
+            Statement:
+              - {Effect: Allow, Action: 'secretsmanager:GetSecretValue', Resource: !Ref DatabaseSecretAdmin}
```
Comment (pablo-code-org) [inline aws/cloudformation/components/database.yml.erb]: I got this code Seems to be easier to read

```
              Action: 
                - 'secretsmanager:GetSecretValue'
              Resource: 
                - !Ref DatabaseSecretAdmin
                - !Ref DatabaseSecretWriter
                - !Ref DatabaseSecretReader
```

**Pull request closed without merging**